docs/content/guides/dagster-pipes/subprocess/modify-external-code.mdx

```diff
@@ -0,0 +1,248 @@
+
+- [Make Dagster context available in external code](#step-1-make-dagster-context-available-in-external-code)
+- [Stream log messages back to Dagster](#step-2-send-log-messages-to-dagster)
+- [Stream structured metadata back to Dagster](#step-3-send-structured-metadata-to-dagster)
+
+---
+
+## Step 1: Make Dagster context available in external code
+
+Getting external code to send information back to Dagster via Dagster Pipes requires adding a few lines of code:
+
+- Imports from `dagster-pipes`
+
+- A call that connects to Dagster Pipes: <PyObject module="dagster_pipes" object="open_dagster_pipes"/> initializes the Dagster Pipes context that can be used to stream information back to Dagster. It is recommended to call this function near the entry point of a pipes session.
+
+  <Note>
+
+  The <code>with open_dagster_pipes():</code> is a context manager in Python, ensuring resource setup and cleanup for a specific segment of code. It's useful for tasks requiring initial setup and final teardown, like opening and closing connections. In this case, the context manager is used to initialize and close the Dagster Pipes connection.
+
+  </Note>
+
+- An instance of the Dagster Pipes context via <PyObject module="dagster_pipes" object="PipesContext" method="get"/>.
+
+In our sample Python script, the changes would look like the following:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_with_context.py lines=2-
+import pandas as pd
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    total_orders = len(orders_df)
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    print(f"processing total {total_orders} orders")
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes():
+        main()
+```
+
+---
+
+## Step 2: Send log messages to Dagster
+
+Dagster Pipes context offers a built-in logging capability that enables you to stream log messages back to Dagster. Instead of printing to the standard output, you can use the `context.log` method on <PyObject module="dagster_pipes" object="PipesContext" /> to send log messages back to Dagster.
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_with_logs.py
+import pandas as pd
+from dagster_pipes import PipesContext, open_dagster_pipes
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    total_orders = len(orders_df)
+    # get the Dagster Pipes context
+    context = PipesContext.get()
+    context.info.log(f"processing total {total_orders} orders")
+
+
+if __name__ == "__main__":
+    # connect to Dagster Pipes
+    with open_dagster_pipes():
+        main()
+```
+
+Then, the log messages will show up in the Dagster UI. You can filter the log levels to only view `info` level messages:
```

Comment:
Kind of a nit - this assumes the user is still on the Runs page. Mind saying it explicitly here? Ex:

> Then, the log messages will show up in the **Run details** page of the Dagster UI.

Reply:
One more thing - users can specify any level of log here, right? Right now this reads like it's only possible to have `info` logs.